homeassistant/helpers/debounce.py

```diff
@@ -104,3 +109,10 @@ def _schedule_timer(self) -> None:
             self.cooldown,
             lambda: self.hass.async_create_task(self._handle_timer_finish()),
         )
+
+    @callback
+    def _sync_job(self) -> None:
+        """Ensure job is in sync with function."""
+        assert self.function is not None
```

Comment:
Maybe we should make `function` a property and use a setter to automatically sync on set ?